harlogger/sniffers.py

```diff
@@ -0,0 +1,218 @@
+import dataclasses
+import json
+import os
+import posixpath
+from abc import ABC, abstractmethod
+from typing import Any, Mapping
+
+import click.types
+from haralyzer import HarEntry
+from pygments import highlight
+from pygments.formatters.terminal256 import TerminalTrueColorFormatter
+from pygments.lexers.textfmts import HttpLexer
+from pymobiledevice3.lockdown import LockdownClient
+from pymobiledevice3.services.os_trace import OsTraceService
+
+
+@dataclasses.dataclass
+class EntryHash:
+    pid: int
+    process_name: str
+    image: str
+    url: str
+
+
+class SnifferStrategy(ABC):
+
+    @abstractmethod
+    def sniff(self) -> None:
+        pass
+
+    @abstractmethod
+    def show(self, entry_hash: EntryHash, entry: Any) -> None:
+        pass
+
+    @abstractmethod
+    def should_keep(self, entry_hash: EntryHash) -> bool:
+        pass
+
+
+class SnifferPreference(SnifferStrategy):
+
+    def __init__(self, lockdown: LockdownClient, filters: Mapping = None, unique: bool = False, response: bool = True,
```

Comment:
- prefix each "show" parameter into `show_`
- for example: `show_unique`, `show_response`